packages/dev-console/src/state/context.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  createContext,
+  useEffect,
+  useContext,
+  useState,
+  useRef,
+  useMemo,
+  useCallback,
+} from 'react';
+
+import {DevServerCall, DevServerResponse} from '../types';
+
+import {Console, ConsoleState, Listener} from './types';
+import {useConsoleReducer, initialConsoleState} from './reducer';
+
+type UnsubscribeFn = () => void;
+
+export interface DevConsoleContextValue {
+  host: string;
+  state: ConsoleState;
```

Comment:
I think it would make it easier to have `extensions` and `app` at the top level. I'm not sure of state adds any value here, but I could be missing something!